Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift

```diff
@@ -299,25 +299,36 @@ open class AVFoundationPlayback: Playback {
             self,
             selector: #selector(playbackDidEnd),
             name: .AVPlayerItemDidPlayToEndTime,
-            object: player.currentItem)
+            object: player.currentItem
+        )
         
         NotificationCenter.default.addObserver(
             self,
             selector: #selector(onAccessLogEntry),
             name: .AVPlayerItemNewAccessLogEntry,
-            object: nil)
+            object: nil
+        )
 
         NotificationCenter.default.addObserver(
             self,
             selector: #selector(onFailedToPlayToEndTime),
             name: .AVPlayerItemFailedToPlayToEndTime,
-            object: nil)
+            object: nil
+        )
+    }
+
+    @objc func playbackDidEnd(notification: NSNotification?) {
+        guard didFinishedItem(from: notification) else { return }
+        trigger(.didComplete)
+        updateState(.idle)
+        droppedFrames = 0
     }
 
     @objc func onFailedToPlayToEndTime(notification: NSNotification?) {
-        if let error = notification?.userInfo?["AVPlayerItemFailedToPlayToEndTimeErrorKey"] as? Error {
-            trigger(.error, userInfo: ["error": error])
-        }
+        let errorKey = "AVPlayerItemFailedToPlayToEndTimeErrorKey"
+        guard let error = notification?.userInfo?[errorKey] as? NSError else { return }
```

Comment:
It is ok do not send any error in case this cast does not work? Maybe would be better to have a default error for this case.

Reply:
Sure.
=]